Classes/ios/UIViewController+MBDebugPanel.m

```diff
@@ -0,0 +1,157 @@
+    }
+}
+
+- (BOOL)mb_debugMenuItemsManagementEnabled
+{
+    BOOL result = NO;
+    
+    id boolObject = objc_getAssociatedObject(self, CFBridgingRetain(MB_UIViewController_Utility_debugMenuItemsManagementEnabledKey) );
+    if ( [boolObject isKindOfClass:[NSNumber class] ] )
+    {
+        NSNumber *boolNumber = boolObject;
+        result = [boolNumber boolValue];
+    }
+    
+    return result;
+}
+
+- (BOOL)viewIsVisible
+{
+    return self.isViewLoaded && self.view.superview != nil;
+}
+
+- (void)mb_addManagedDebugMenuItems:(NSArray *)items
+{
+    if (self.mb_managedDebugMenuItems == nil)
+    {
+        self.mb_managedDebugMenuItems = [NSMutableArray arrayWithArray:items];
+    } else
+    {
+        [self.mb_managedDebugMenuItems addObjectsFromArray:items];
+    }
+    
+    if (self.mb_debugMenuItemsManagementEnabled && self.viewIsVisible)
+    {
+        [MBDebugPanel addComponentsFromArray:items];
+    }
+}
+
+- (void)mb_removeManagedDebugMenuItems:(NSArray *)items
+{
+    [self.mb_managedDebugMenuItems removeObjectsInArray:items];
+}
+
+- (void)setMb_managedDebugMenuItems:(NSMutableArray *)managedDebugMenuItems
+{
+    objc_setAssociatedObject(self, CFBridgingRetain(MB_UIViewController_Utility_managedDebugMenuItemsKey), managedDebugMenuItems, OBJC_ASSOCIATION_RETAIN);
+}
+
+- (NSMutableArray *)mb_managedDebugMenuItems
+{
+    NSMutableArray *result;
+    
+    id itemsObject = objc_getAssociatedObject(self, CFBridgingRetain(MB_UIViewController_Utility_managedDebugMenuItemsKey) );
+    if ( [itemsObject isKindOfClass:[NSMutableArray class] ] )
+    {
+        result = itemsObject;
+    }
+    
+    return result;
+}
+
+- (void)mb_addDebugMenuGestureRecognizer:(UIGestureRecognizer *)recognizer
+{
+    [self mb_addDebugMenuGestureRecognizer:recognizer onView:self.view];
+}
+
+- (void)mb_addDebugMenuGestureRecognizer:(UIGestureRecognizer *)recognizer onView:(UIView *)view
+{
+    [recognizer addTarget:self action:@selector(mb_openDebugPanelFromGesture:) ];
+    [view addGestureRecognizer:recognizer];
+}
+
+- (void)mb_addDebugMenuTapGestureRecognizer:(NSUInteger)numberOfTaps
+{
+    [self mb_addDebugMenuTapGestureRecognizer:numberOfTaps onView:self.view];
+}
+
+- (void)mb_addDebugMenuTapGestureRecognizer:(NSUInteger)numberOfTaps onView:(UIView *)view
+{
+    UITapGestureRecognizer *tapGesture = [ [UITapGestureRecognizer alloc] initWithTarget:self
+                                                                                  action:@selector(mb_openDebugPanelFromGesture:) ];
+    tapGesture.numberOfTapsRequired = numberOfTaps;
+    [view addGestureRecognizer:tapGesture];
+}
+
+- (void)mb_openDebugPanel
+{
+    [MBDebugPanel show];
+}
+
+- (void)mb_openDebugPanelFromGesture:(UITapGestureRecognizer *)tapGesture
+{
+    [MBDebugPanel show];
+}
+
```

Comment:
What is the purpose of these final two methods?  (Since they simply call the already-public `[MBDebugPanel show]`.)    Is there a reason for client code to not just call `[MBDebugPanel show]` directly?